pkg/chunkenc/memchunk.go

```diff
@@ -1166,22 +1165,25 @@ func (si *bufferedIterator) Next() bool {
 
 // moveNext moves the buffer to the next entry
 func (si *bufferedIterator) moveNext() (int64, []byte, bool) {
-	ts, err := binary.ReadVarint(si.bufReader)
-	if err != nil {
-		if err != io.EOF {
-			si.err = err
-		}
-		return 0, nil, false
-	}
-
-	l, err := binary.ReadUvarint(si.bufReader)
-	if err != nil {
-		if err != io.EOF {
-			si.err = err
-			return 0, nil, false
+	var ts int64
+	var tWidth, lWidth, lineSize int
+	for lWidth == 0 { // Read until both varints have enough bytes.
```

Comment:
I realise this can loop forever on some corrupt inputs - if there are some bytes in the buffer but not enough to decode two varints. Need to detect this somehow and error.